File: tools/scale_hotspots.py
#!/usr/bin/env python3
"""
Скрипт для пересчёта hotspot координат из 960×640 в 1280×720
"""

import re
from pathlib import Path

SCALE_X = 1280 / 960  # 1.333
SCALE_Y = 720 / 640   # 1.125

def scale_rect(x, y, w, h):
    """Пересчитывает координаты прямоугольника"""
    return (
        round(x * SCALE_X),
        round(y * SCALE_Y),
        round(w * SCALE_X),
        round(h * SCALE_Y)
    )

def replace_rect(match):
    """Заменяет rect = { x = ..., y = ..., w = ..., h = ... }"""
    x = int(match.group(1))
    y = int(match.group(2))
    w = int(match.group(3))
    h = int(match.group(4))
    
    new_x, new_y, new_w, new_h = scale_rect(x, y, w, h)
    
    return f"rect = {{ x = {new_x}, y = {new_y}, w = {new_w}, h = {new_h} }}"

def main():
    scenes_path = Path("main/scripts/scenes.lua")
    
    if not scenes_path.exists():
        print(f"Error: {scenes_path} not found")
        return
    
    # Читаем файл
    with open(scenes_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Обновляем комментарий
    content = content.replace(
        "в коорд. системе .gui (960×640",
        "в коорд. системе .gui (1280×720"
    )
    
    # Заменяем все rect = { x = ..., y = ..., w = ..., h = ... }
    pattern = r'rect = \{ x = (\d+), y = (\d+), w = (\d+), h = (\d+) \}'
    content = re.sub(pattern, replace_rect, content)
    
    # Сохраняем
    with open(scenes_path, 'w', encoding='utf-8') as f:
        f.write(content)
    
    print(f"[OK] Updated {scenes_path}")
    print(f"  Scale factors: X={SCALE_X:.3f}, Y={SCALE_Y:.3f}")
    print(f"  All hotspot rectangles have been recalculated")

if __name__ == "__main__":
    main()

File: tools/scale_hotspots.lua
-- scale_hotspots.lua
-- Скрипт для пересчёта hotspot координат из 960×640 в 1280×720
-- Использование: lua tools/scale_hotspots.lua

local SCALE_X = 1280 / 960  -- 1.333
local SCALE_Y = 720 / 640   -- 1.125

local function scale_rect(rect)
    return {
        x = math.floor(rect.x * SCALE_X + 0.5),
        y = math.floor(rect.y * SCALE_Y + 0.5),
        w = math.floor(rect.w * SCALE_X + 0.5),
        h = math.floor(rect.h * SCALE_Y + 0.5)
    }
end

local function scale_position(x, y)
    return math.floor(x * SCALE_X + 0.5), math.floor(y * SCALE_Y + 0.5)
end

-- Читаем scenes.lua
local file = io.open("main/scripts/scenes.lua", "r")
if not file then
    print("Error: Cannot open main/scripts/scenes.lua")
    os.exit(1)
end

local content = file:read("*all")
file:close()

-- Обновляем комментарий в начале файла
content = content:gsub(
    "rect = { x, y, w, h } %-%- x/y это ЛЕВЫЙ%-НИЖНИЙ угол прямоугольника\n%-%- в коорд%. системе %.gui %(960×640",
    "rect = { x, y, w, h } -- x/y это ЛЕВЫЙ-НИЖНИЙ угол прямоугольника\n-- в коорд. системе .gui (1280×720"
)

-- Функция для замены rect = { x = ..., y = ..., w = ..., h = ... }
local function replace_rect(match)
    local x, y, w, h = match:match("x = ([%d%.]+), y = ([%d%.]+), w = ([%d%.]+), h = ([%d%.]+)")
    if x and y and w and h then
        local old_rect = {x = tonumber(x), y = tonumber(y), w = tonumber(w), h = tonumber(h)}
        local new_rect = scale_rect(old_rect)
        return string.format("x = %d, y = %d, w = %d, h = %d", 
            new_rect.x, new_rect.y, new_rect.w, new_rect.h)
    end
    return match
end

-- Заменяем все rect = { ... }
content = content:gsub("rect = { (x = [^}]+) }", function(inner)
    return "rect = { " .. replace_rect(inner) .. " }"
end)

-- Сохраняем результат
local out = io.open("main/scripts/scenes.lua", "w")
out:write(content)
out:close()

print("✓ Updated main/scripts/scenes.lua")
print("  Scale factors: X=" .. SCALE_X .. ", Y=" .. SCALE_Y)
print("  All hotspot rectangles have been recalculated")
